src/main/daml/Daml/Finance/Interface/Types/Fpml.daml

```diff
@@ -20,31 +22,78 @@ data SwapStream = SwapStream
 -- | The calculation periods dates schedule.
 data CalculationPeriodDates = CalculationPeriodDates
   with
-    -- id : Text
-    effectiveDate : Date
-    terminationDate : Date
+    id : Text
+    effectiveDate : AdjustableDate
+    terminationDate : AdjustableDate
     calculationPeriodDatesAdjustments : CalculationPeriodDatesAdjustments
     firstRegularPeriodStartDate : Optional Date
     lastRegularPeriodEndDate : Optional Date
     calculationPeriodFrequency : CalculationPeriodFrequency
   deriving (Eq, Show)
 
+-- | The payment dates schedule.
+data PaymentDates = PaymentDates
+  with
+    -- id : Optional Text
+    calculationPeriodDatesReference : Text
+    paymentFrequency : PaymentFrequency
+    firstPaymentDate : Optional Date
+    lastRegularPaymentDate : Optional Date
+    payRelativeTo : DateRelativeToEnum
+    paymentDatesAdjustments : BusinessDayAdjustments
+  deriving (Eq, Show)
+
+-- | A type for defining a date that shall be subject to
+--   adjustment if it would otherwise fall on a day that
+--   is not a business day in the specified business
+--   centers, together with the convention for adjusting
+--   the date.
+data AdjustableDate = AdjustableDate with
+  -- id_ : Optional Text
+  unadjustedDate : Date
+      -- ^ A date subject to adjustment.
+  dateAdjustments : BusinessDayAdjustments
+      -- ^ The business day convention and financial business
+      --   centers used for adjusting the date if it would
+      --   otherwise fall on a day that is not a business date
+      --   in the specified business centers.
+  -- adjustedDate : Optional IdentifiedDate
+      -- ^ The date once the adjustment has been performed.
+      --   (Note that this date may change if the business
+      --   center holidays change).
+    deriving (Eq, Show)
+
+-- | A type defining the business day convention and
+--   financial business centers used for adjusting any
+--   relevant date if it would otherwise fall on a day
+--   that is not a business day in the specified business
+--   centers.
+data BusinessDayAdjustments = BusinessDayAdjustments with
+  -- id_ : Optional Text
+  businessDayConvention : BusinessDayConventionEnum
+      -- ^ The convention for adjusting a date if it would
+      --   otherwise fall on a day that is not a business day.
+  businessCenters : Optional [Text]
+    deriving (Eq, Show)
+
 -- | The reset dates schedule. This only applies for a floating rate stream.
 data ResetDates = ResetDates
   with
-    resetRelativeTo : ResetRelativeToEnum
+    calculationPeriodDatesReference : Text
+    resetRelativeTo : DateRelativeToEnum
     fixingDates : FixingDates
-    -- resetFrequency : ResetFrequency
+    resetFrequency : ResetFrequency
+    resetDatesAdjustments : ResetDatesAdjustments
   deriving (Eq, Show)
 
--- | The specification of whether resets occur relative to
+-- | The specification of whether payments/resets occur relative to
```

Comment:
Could you remind me what are the reset dates? Are they accrual dates?

Reply:
The resetDates are the dates where the floating rate is fixed (or "reset") for each period:
https://www.fpml.org/spec/fpml-5-11-3-lcwd-1/html/confirmation/schemaDocumentation/schemas/fpml-ird-5-11_xsd/complexTypes/InterestRateStream/resetDates.html